src/main/kotlin/pl/droidsonroids/gradle/jacoco/testkit/JaCoCoTestKitPlugin.kt

```diff
@@ -2,17 +2,43 @@ package pl.droidsonroids.gradle.jacoco.testkit
 
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
 import org.gradle.testing.jacoco.plugins.JacocoPlugin.DEFAULT_JACOCO_VERSION
 import org.gradle.testing.jacoco.plugins.JacocoPluginExtension
-import pl.droidsonroids.gradle.jacoco.testkit.Configurations.currentTestRuntime
+import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
 import pl.droidsonroids.gradle.jacoco.testkit.Configurations.jacocoRuntime
-import pl.droidsonroids.gradle.jacoco.testkit.Tasks.generateJacocoTestKitProperties
-import pl.droidsonroids.gradle.jacoco.testkit.Tasks.test
+
+open class JaCoCoTestKit(private val project: Project) {
+    val jacocoRuntimePathProvider: Provider<String> = project.provider { project.configurations.getByName(jacocoRuntime).asPath }
+
+    fun applyTo(configurationRuntime: String, taskProvider: TaskProvider<Task>) {
+        with(project) {
+            dependencies.add(configurationRuntime, files(testKitDir()))
+            val jacocoTestKitPropertiesTask = tasks.register(
+                "generateJacoco${taskProvider.name.capitalize()}KitProperties",
```

Comment:
We should probably add `Locale.ENGLISH` argument to `capitalize()`.
I haven't checked that in practice but It seems that if task name starts with `i` and default locale at run time is Turkish or eg. Azeri then argumentless capitalize outputs non-ASCII capital dotted `İ`.

Reply:
It forces me to do `@ExperimentalStdlibApi`

Reply:
Hmm, ok let's leave it as is for now. I'll check that deeper.

Reply:
As that function was added in Kotlin 1.3 it means the plugin will require Gradle 5.6 (kotlin was updated on 5.5 but it doesn't work until 5.6 for some reason)